### ISSUE TITLE
Hack for openSUSE

### ISSUE THREAD

**Issue by wismill** [david-janssen/kmonad#4]
Hi there, thanks for the great work!

I had trouble to make kmonad work on openSUSE Tumbleweed. It does work using `sudo`, but if I want to run kmonad as normal user on my system there is no such group `uinput`. There is a group `input` but simply adding it to my user does not solve the issue.

My current solution is:
- create a group `uinput` and add my user to it
- add my user to the group `input`
- create a file `/lib/udev/rules.d/99-kmonad.rules` with the following content:
      ```
      KERNEL=="uinput", MODE="0660", GROUP="uinput", OPTIONS+="static_node=uinput"
      ```
- reboot

I am no expert. I wonder if this is secure and if it could be simplified?

**Comment by slotThe** [comment]
I don't see why this would not be secure, afaik creating a group is the standard way to give users access to kernel modules (If this is wrong I would highly appreciate someone correcting me here).

As for simplifying, you don't have to create a `uinput` group if you already have `input`. I'm not sure what `OPTIONS+="static_node=uinput"` does but I'm fairly certain `MODE="0600"` is unecessary.  My `.rules` file simply contains `KERNEL=="uinput", GROUP="input"`, which seems to work fine so far.  Note that this is all from the perspective of a Void Linux install, I sadly don't have a Tumbleweed box at hand right now to test these wild speculations of mine.

**Comment by wismill** [comment]
I removed the `uinput` group I created and change the rules files accordingly, but it requires the `OPTIONS` field to work:
```
KERNEL=="uinput", MODE="0660", GROUP="input", OPTIONS+="static_node=uinput"
```

**Comment by david-janssen** [comment]
I'm happy you managed to get things working. This isn't really a KMonad issue in the sense that KMonad needs to be adjusted to make this work, but I think it might be a good idea to create some documentation on how to get things running for different distros.

Should I just create an additional markdown file in the `doc` folder with further instructions for distros? You guys have an idea/opinion?

**Comment by wismill** [comment]
Totally agree. Keep it simple, add a file and a link to it in the readme. This doc could then be completed with users' feedbacks from different distros.

**Comment by david-janssen** [comment]
Very good. Won't have time for a few days, but will have a go in a couple of days. Once I do I'll close this issue.

**Comment by slotThe** [comment]
I'm not sure creating a separate file is really necessary. I vote for just adding a note to the README, telling users to create a udev rule with
```
KERNEL=="uinput", MODE="0660", GROUP="input", OPTIONS+="static_node=uinput"
```
This seems to work on a lot of distros (in addition to OPs openSUSE install I have just tested this on Void and Arch). Should there be any more problems I suppose we can then put it into an extra file.

I have created #5 as a first draft of how I would instead change just the README.
